test/lib/y2firewall/clients/installation_finish_test.rb

```diff
@@ -47,48 +42,90 @@
       let(:installed) { false }
 
       it "returns true" do
+        expect(subject).to_not receive(:configure_firewall)
         expect(subject.write).to eq true
       end
     end
 
     context "when firewalld is installed" do
-      it "enables the firewalld service if enabled in the proposal" do
-        allow(proposal_settings).to receive("enable_firewall").and_return(true)
-        expect(firewalld).to receive("enable!")
+      it "configures the firewall according to the proposal settings" do
+        expect(subject).to receive(:configure_firewall)
 
         subject.write
       end
 
-      it "disables the firewalld service if disabled in the proposal" do
-        expect(firewalld).to receive("disable!")
-
-        subject.write
+      it "returns true" do
```

Comment:
I'd expect more verbose test description

Reply:
IMHO it looks good. @teclator decided to write one "it" section for each thing he wants to test. If you check the previous one, you can read  *configures the firewall according to the proposal settings* (which is more verbose :smiley:). And this one is just about making sure that the method returns the right value.

Reply:
if you imagine test overview, you'll get something like
```
When firewalld is installed
...
  returns true
```
and it doesn't look that good in my POV

Reply:
Exactly ;)

Reply:
@imobachgs exactly

Reply:
@mchf and it describes explicitly what the methods returns without confusion. I decided to put inside the context because from my POV it describes better that without firewalld installed it does nothing special and shows that the scenarios is covered.

So, which description would you prefer?:

```
#write
  enables the sshd service if enabled in the proposal
  when firewalld is not installed
    returns true
  when firewalld is installed
    configures the firewall according to the proposal settings
    returns true
```

```
#write
  enables the sshd service if enabled in the proposal
  returns true
  when firewalld is installed
    configures the firewall according to the proposal settings
```

Reply:
the first one ;-) But anyway this is NP for me.